plugins/parquet/parquetembed.cpp

```diff
@@ -1015,14 +1020,10 @@ double ParquetRowBuilder::getRealResult(const RtlFieldInfo *field)
         return p.doubleResult;
     }
 
-    switch((*array_visitor)->type)
-    {   case DoubleType:
-            return (*array_visitor)->double_arr->Value(currArrayIndex());
-        case FloatType:
-            return (*array_visitor)->float_arr->Value(currArrayIndex());
-        default:
-            return getCurrValue(field);
-    }
+    if ((*array_visitor)->type == RealType)
+        return getReal(array_visitor, currArrayIndex());
+    else
+        return getCurrIntValue(field);
```

Comment:
Strings will not be converted to real values.  However I would be tempted to create a separate PR to fix that and merge this one as-is.